fluent/client/ws_client_test.go

```diff
@@ -303,6 +306,48 @@ var _ = Describe("WSClient", func() {
 			Expect(bytes.Equal(bits, writtenbits)).To(BeTrue())
 		})
 
+		It("Sends the message", func() {
+			msgBytes, _ := msg.MarshalMsg(nil)
+			Expect(client.Send(&msg)).ToNot(HaveOccurred())
+
+			writtenBytes := conn.WriteArgsForCall(0)
+			Expect(bytes.Equal(msgBytes, writtenBytes)).To(BeTrue())
+		})
+
+		When("The message is large", func() {
+			const charset = "abcdefghijklmnopqrstuvwxyz" + "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+
+			var (
+				expectedBytes int
+				writtenBytes  int
+				b             bytes.Buffer
+			)
+
+			BeforeEach(func() {
+				messageSize = 65536
+			})
+
+			JustBeforeEach(func() {
+				seededRand := rand.New(
+					rand.NewSource(time.Now().UnixNano()))
+				m := make([]byte, messageSize)
+				for i := range m {
+					m[i] = charset[seededRand.Intn(len(charset))]
+				}
+				msg.Record = m
+
+				Expect(client.Send(&msg)).ToNot(HaveOccurred())
+				Expect(msgp.Encode(&b, &msg)).ToNot(HaveOccurred())
+				Expect(conn.WriteCallCount()).To(Equal(1))
+				expectedBytes = len(b.Bytes())
+				writtenBytes = len(conn.WriteArgsForCall(0))
```

Comment:
This code is still mixing preparation and test. It's hard for a newbie to understand what is to be tested - at least from my point of view.

Writes to `bytes.Buffer()` are used to calculate the size that is expected in the following test.

So:
* lines 340 and 342 are preparation and should remain in `JustBeforeEach()`. Thus also `bytes.Buffer()` gets a helper which doesn't need to survive the scope of `JustBeforeEach()`
* Lines 341 and 343 are testing the function and thus should be placed in the `It` section.
* Line 339 can be discussed if in `JustBeforeEach()` or in `It()`. For newbies it's probably easier to move it also to `It()` but if it remains in `JustBeforeEach()` place it at the very end because that's the code that is to be proven in `It()`